protocol/eth-shared-lockbox.md

```diff
@@ -0,0 +1,192 @@
+
+### Unified Chain Governance
+
+OP Chains will be governed within a common Chain Cluster governance entity (the Collective). Currently, this entity is responsible for:
+
+- **Ensuring chains are consistent at the implementation side**, achieved either through trusted deployment methods (e.g., OP Contracts Manager) or by approval after security checks are performed.
+- **Approving protocol upgrades** for all chains involved.
+- **Adding new chains** to the interoperable set, and managing the `dependencyManager` role. The final decision is up to governance and chains cannot be removed afterwards.
+- **Replacing chain servicers** for existing chains if they fail to satisfy technical requirements.
+
+### Shared Bridging and SuperchainWETH usage
+
+In any OP Chain that joins the cluster, it is assumed that `SuperchainWETH` has been deployed in advance prior to being added to the interoperable graph. As a result, the equivalence between ETH deposits and withdrawal history and the actual ETH supply will vary from the outset. In a world with freedom of movement, all real ETH liquidity is theoretically shared across the entire cluster eventually, regardless of how deposits are handled.
+
+# Solution
+
+The existing problem and considerations motivate us to propose an L1 shared liquidity design through the introduction of a new `SharedLockbox` on L1, which serves as a singleton contract for ETH, given a defined set of interoperable chains. To ensure consistency, the `SuperchainConfig` is extended to act as the *dependency set manager*, controlling the op-governed dependency set that the lockbox uses as a source of truth. New ETH deposits will be directed to the lockbox, with the same process applied to ETH withdrawals.
+
+### Spec changes
+The core proposed changes are as follows:
+- **Modify the `SuperchainConfig` contract**: Extend this contract to manage the dependency set of each chain, taking ownership of the `dependencyManager` role for each `SystemConfig`.
+- **Introduce the `SharedLockbox` contract**: This contract acts as an escrow for ETH, receiving deposits and allowing withdrawals from approved `OptimismPortal` contracts. The `SuperchainConfig` contract serves as the source of truth of the lockbox.
+- **Modify the `OptimismPortal`**: To forward ETH into the `SharedLockbox` when `depositTransaction` is called, with the inverse process applying when `finalizeWithdrawal` is called.
+
+### Managing op-governed dependency set through `SuperchainConfig`
+
+The `SuperchainConfig` contract will serve as the single point for managing the dependency set of a cluster and will be managed by the admin as occurs in other L1 OP contracts. This contract assumes the role of `dependencyManager` for every `SystemConfig` contract involved. Since the dependency graph follows a simple dependency, it only stores a mapping (or array) of chains added, e.g. given a `chainId`.
```

Comment:
We could simplify and remove the concept of `dependencyManager` and instead just hardcode it to be  the `SuperchainConfig`. Unclear what the tradeoffs are here

Reply:
The change involves hardcoding the current L1 `SuperchainConfig` address in the `SystemConfigInterop` contract and removing `dependencyManager` as an `initialize` parameter.

Pros:
- Reduces code complexity.
- Simplifies the migration process by minimizing steps, avoiding the need to reset the `initialized` slot.

Cons:
- Reduces flexibility.
- New chains deployed through the OPCM will share the same `SuperchainConfig`, which might not be desirable for clusters outside the Superchain.

About cons, we always tend to think about future clusters but we are not sure if we should do it and we will like to know you vision on this

Reply:
Lets enshrine the `SuperchainConfig` rather than having a `dependencyManager`, this will enforce the assumptions in https://github.com/ethereum-optimism/specs/issues/460 and make the core product always be a highly connected graph. It makes no sense having another topology until transitive deps don't need to be ran to fully validate the network. This also lets us remove a role from the `SystemConfig`, we may need to move it to the `SuperchainConfig`

Reply:
Depending on the design of OPCM, it may mean that there is 1 OPCM per cluster, if the `SuperchainConfig` is `immutable`